django/contrib/admin/options.py

```diff
@@ -884,7 +884,11 @@ def construct_search(field_name):
             if field_name.startswith('^'):
                 return "%s__istartswith" % field_name[1:]
             elif field_name.startswith('='):
-                return "%s__iexact" % field_name[1:]
+                field_name = field_name[1:]
+                # Integers don't care about case-sensitivity.
+                if isinstance(self.model._meta.get_field(field_name), models.IntegerField):
+                    return "%s__exact" % field_name
+                return "%s__iexact" % field_name
```

Comment:
It's a shame we register the `__iexact` lookup for all fields. I think `not isintance(..., (models.CharField, models.TextField))` would be more appropriate here.